department-of-veterans-affairs/caseflow
# SeanC/APPEALS-9312

Resolves [APPEALS-9312](https://vajira.max.gov/browse/APPEALS-9312)

### Description
Continues to distribute AMA cases to judges even if the connection to VACOLS is lost

### Acceptance Criteria
- [ ] Job doesn't stop running when VACOLS connection is lost during distribution
- [ ] Continues to distribute AMA cases

### Testing Plan
1. Make sure there are AMA and Legacy cases to distribute (see notes below for commands to create cases)
2. Go to Scheduled Jobs tab in Caseflow
3. Shut down VACOLS
4. Run PushPriorityAppealsToJudgesJob
5. Once the job has completed start VACOLS
6. Create more AMA and Legacy cases to distribute
7. Run the job again from the Scheduled Jobs tab
8. Shut down VACOLS while the job is running
9. Once the job has completed start VACOLS again
10. Create more AMA and Legacy cases to distribute
11. Run the job again from the Scheduled Jobs tab

Notes (you can change how many cases the command create):
- Command to create AMA cases: 
```
200.times do
  FactoryBot.create(:appeal, :direct_review_docket, :ready_for_distribution, :advanced_on_docket_due_to_age)
end
```
- Command to create Legacy cases:
```
200.times do
  FactoryBot.create(:case, :aod, :ready_for_distribution, :type_original, correspondent: FactoryBot.create(:correspondent))
end
```


### User Facing Changes

 BEFORE|AFTER
 ---|---

### Code Documentation Updates

### Database Changes
*Only for Schema Changes*


### Integrations: Adding endpoints for external APIs
* [ ] Integrations impacting functionality are tested in Caseflow UAT
